### ISSUE TITLE
error: flake '...' does not provide attribute '...activationPackage'

### ISSUE THREAD

**Issue by jan-senaa** [t184256/nix-on-droid#231]
Hi. Today I've tried making a new build with flakes, and I've come across the following error:

```bash
$ nix flake init --template github:t184256/nix-on-droid#minimal
$ nix-on-droid switch --flake .

Building activation package...
error: flake 'path:/data/data/com.termux/nix/files/home/minimal' does not provide attribute 'packages.aarch64-linux.nixOnDroidConfigurations...activationPackage', 'legacyPackages.aarch64-linux.nixOnDroidConfigurations...activationPackage' or 'nixOnDroidConfigurations...activationPackage'
```

I've tried with all the [templates](https://github.com/t184256/nix-on-droid/tree/master/templates) in this repository, and plus my custom configs. None worked.

My current config (without the flakes) is also pretty minimal and contains nothing but `vim`, `git` and flakes experimental settings:

```nix
{ pkgs, ... }:

{
  environment.packages = with pkgs; [ vim git ];
  nix.extraOptions = ''
    experimental-features = nix-command flakes
  '';
  system.stateVersion = "22.05";
}
```

A fresh installation of the app didn't change anything as well.

**Comment by jan-senaa** [comment]
Thank you so much for your attention. It indeed works with the latest bootstrap that you've provided. Please forgive my reckless behaviour. I should've checked the older branches' README first. I'm so sorry for stealing your time for non-important stuff.

But the templates still have a tiny little problem that I would like to mention:

The templates use the `release-22.05` branch as `nix-on-droid` input, which requires the `config` argument, but the code in the template uses `modules` argument instead of `config`. This causes the following problem.

```bash
error: 'nixOnDroidConfiguration' at /nix/store/wn8d0db2zmskrsnvhvzqcik0cn3v9x9h-source/flake.nix:29:9 called without required argument 'config'

    at /nix/store/m9zdv704p6jq1n1bff7qx4n0jk1wq953-source/flake.nix:21:40:

        20|
        21|    nixOnDroidConfigurations.default = nix-on-droid.lib.nixOnDroidConfiguration {
          |                                       ^
        22|      modules = [
```

This doesn't occur when the input branch is `master` instead of `release-22.05`. I guess this isn't that urgent and people can fix it themselves maybe, but I wanted to point it out anyways.

Again, thank you so much for your help. I'm so sorry for bothering you for nothing. My issue has been solved.

**Comment by Gerschtli** [comment]
No worries :)

Oh that's a good catch. But because the release of 21.11 will happen in a few days, we can probably wait for it and then bump all usages of 22.05 to 22.11 and then it just works.

**Comment by t184256** [comment]
My bad. I've updated https://nix-on-droid.unboiled.info/bootstrap/bootstrap-aarch64.zip, which had been previously updated on 2022-10-03.

**Comment by Gerschtli** [comment]
What do you think about uploading latest zipball as part of the pipeline to reduce manual stuff like this? @t184256

**Comment by t184256** [comment]
I'm not comfortable with granting permissions for that to pipelines.

**Comment by Gerschtli** [comment]
Fixed with #234
